staging/src/k8s.io/component-helpers/cdi/validation/validation.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateCDIDriverName checks that a string meets the requirements for the
+// ResourceClass.DriverName field. The requirements are the same as for a CSI
+// driver name.
+func ValidateDriverName(driverName string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(driverName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, ""))
+	}
+
+	if len(driverName) > 63 {
```

Comment:
Why 63? In this case a constant or variable with a meaningful name would be more readable I believe.

Reply:
This is the same length limit as for CSI driver name. The entire function was copied from there.

I am following how API validation is handled elsewhere in Kubernetes:
- not all limitations are fully documented in code comments (but perhaps should be - not sure)
- no constants for such limits, neither in the API nor in the code

I can imagine that defining this limit as part of the API would be useful. I need to check with the API reviewers whether that is a good or bad idea.

Reply:
I've added a TODO.